### ISSUE TITLE
Bug in test_number.py:LongTests.test_2 may be undesired behavior in function long_to_bytes

### ISSUE THREAD

**Issue by mahmoudimus** [Legrandin/pycryptodome#519]
Hi there - I think there's a possible [bug in the tests](https://github.com/Legrandin/pycryptodome/blob/master/lib/Crypto/SelfTest/Util/test_number.py#L152) which might be a bigger bug somewhere else?

```python3
# in python3
>>> from Crypto.Util.number import long_to_bytes
>>> long_to_bytes(0x100, 1) == b'\x01\x00'
True
```

This is correct, but the test is actually asking to see if `long_to_bytes(0x100, 1)` can return `0x100` in *1* byte, which is impossible, since 255 is the maximum number in one byte. Instead of throwing an error, it is returning `b'\x01\x00'`, which is obviously two bytes. 

Python actually does the right behavior:

```python3
>>> int(0x100).to_bytes(1, "big")
----> 1 int(0x100).to_bytes(1, "big")

OverflowError: int too big to convert
```

Am I misunderstanding the test and the function call? 

Thank you!

**Comment by Legrandin** [comment]
The second parameter (`blocksize`) is not the maximum length of the output (like with `to_bytes`) but the minimum padding that should be added. The output can still be longer than that.

**Comment by mahmoudimus** [comment]
@Legrandin - an I see. I think I was confused then, given the comments here: https://github.com/Legrandin/pycryptodome/blob/5dace638b70ac35bb5d9b565f3e75f7869c9d851/lib/Crypto/Util/number.py#L378-L389 - I figured we were emulating `int.to_bytes()` but I guess we're not. I'm happy to close this issue, but maybe we should fix the docstring? I'm happy to make a PR after thinking about how to re-word it?

**Comment by Legrandin** [comment]
Agree, the current documentation is not accurate. I am about to fix it.

**Comment by Legrandin** [comment]
Fixed in version 3.10.4 (https://www.pycryptodome.org/en/latest/src/util/util.html#Crypto.Util.number.long_to_bytes).
